### PR TITLE
docs: Add documentation for `config`

### LINKED ISSUE #8: Add Unit Tests, Comments, and Documentation/Docstrings
Most files need unit tests, comments, and documentation/typing added to them.

Current unit tests are located in `.tests`.

- Docstrings in `rpc` are mostly done, these are all base level implementations of the miner APIs, although some standardization across files and possibly improved type hinting would be useful.
- Unit tests for `network` have been written, but could always be better.
- Generally all files need some level of comments added, in my opinion it is a good idea to be as descriptive as possible, as an end goal is for even novice programmers to be able to design a simple program with this repo.

### TODO:
- #### Add unit tests:
    - [ ]  `rpc`  - This may be tough, these rely on external connections heavily.
    - [ ] `ssh` - Extension of `rpc` essentially, this handles more external calls.
    - [ ] `web` - Extension of `rpc`, again, more external calls.
    - [ ]  `config` - Partially done, but could use more tests for different conversions.
    - [ ]  `miners`  - Same issue as `rpc`.  Partially done, with checks for initialization and data locations.
    - [x] `network` - The important parts here are done, such as making sure inputs turn into the right network values.
- #### Add docstrings/documentation:
    - [x] `rpc`  - Maybe slight improvements to standardization, but overall good.
    - [ ] `web` - A part of data collection alongside `rpc`.  Could be used in the documentation alongside `rpc` in the advanced section.
    - [ ] `ssh` - Another means of data collection.  Same idea, this is an advanced topic.
    - [ ] #124
    - [x]  `miners` - Auto generated with `generate_miners.py`, this generates the supported list and the files for each.
    - [x] `network` - Quite good.  Easy to follow, short, and simple.
    - [x] `settings` - Good enough, doesn't need a whole lot of mentions unless something changes.
- #### Additional Comments:
    - Overall the docs could use a restructure, I'm not sure what the best way to do this is, but things are a bit jumbled right now.

### LINKED ISSUE #124: `config`  - Only source code is shown under the Miner Config page, ideally it would have some descriptions of how to use, along with docstrings from the methods.  It would also be ideal to add documentation on the sub-items.

### PR DESCRIPTION
**Pull request by 1e9abhi1e10**
Partial fix #8 
Fixes #124
Add documentation for `config`

### CODE REVIEW

File: pyasic/config/__init__.py [hunk 1]
```diff
@@ -34,9 +36,11 @@ class MinerConfig:
     )
 
     def as_dict(self) -> dict:
+        """Converts the MinerConfig object to a dictionary."""
         return asdict(self)
 
     def as_am_modern(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for AM Modern miners."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:43]: AM modern just means Antminer modern, you can change this to `...modern Antminers.`

File: pyasic/config/__init__.py [hunk 2]
```diff
@@ -47,6 +51,7 @@ def as_am_modern(self, user_suffix: str = None) -> dict:
         }
 
     def as_wm(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for WM miners."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:54]: Same here, Whatsminers, so `...for Whatsminers`.

File: pyasic/config/__init__.py [hunk 3]
```diff
@@ -56,6 +61,7 @@ def as_wm(self, user_suffix: str = None) -> dict:
         }
 
     def as_am_old(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for old versions of AM miners."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:64]: Same fix as the modern Antminers.

File: pyasic/config/__init__.py [hunk 4]
```diff
@@ -74,6 +81,7 @@ def as_goldshell(self, user_suffix: str = None) -> dict:
         }
 
     def as_avalon(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for Avalon miners."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:84]: Usually stylized as 1 word, `Avalonminers`.

File: pyasic/config/__init__.py [hunk 5]
```diff
@@ -83,6 +91,7 @@ def as_avalon(self, user_suffix: str = None) -> dict:
         }
 
     def as_inno(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for Inno miners."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:94]: These are Innosilicon miners, you can replace `Inno` with `Innosilicon`.

File: pyasic/config/__init__.py [hunk 6]
```diff
@@ -92,6 +101,7 @@ def as_inno(self, user_suffix: str = None) -> dict:
         }
 
     def as_bosminer(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for Bosminer."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:104]: This gets put into a file on the device, but is for the same type of device as the Boser function, just slightly older.  I would prefer to have this be `Generates the configuration in the bosminer.toml format.`

File: pyasic/config/__init__.py [hunk 7]
```diff
@@ -100,6 +110,7 @@ def as_bosminer(self, user_suffix: str = None) -> dict:
         }
 
     def as_boser(self, user_suffix: str = None) -> dict:
+        """"Generates the configuration in the format suitable for Boser."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:113]: Usually stylized as `BOSer`.

File: pyasic/config/__init__.py [hunk 8]
```diff
@@ -109,6 +120,7 @@ def as_boser(self, user_suffix: str = None) -> dict:
         }
 
     def as_epic(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for Epic miners."""
```
Comment (UpstreamData) [inline pyasic/config/__init__.py:123]: Usually stylized as `ePIC`.

**Review by UpstreamData — CHANGES_REQUESTED** [review]
Good start, thanks for this!  Lots of small style things, easier to fix now before this is added to more files.

**Comment by 1e9abhi1e10** [comment]
> Good start, thanks for this! Lots of small style things, easier to fix now before this is added to more files.

I will implement the modifications as per your suggestions.

Also, 
https://github.com/UpstreamData/pyasic/blob/2815d2ba1126c13e711a56feaae15ac26bffae13/pyasic/config/__init__.py#L128-L129

Should I also document these methods?

**Comment by b-rowan** [comment]
Sure, can't hurt.

File: pyasic/config/__init__.py [hunk 9]
```diff
@@ -56,6 +61,7 @@ def as_wm(self, user_suffix: str = None) -> dict:
         }
 
     def as_am_old(self, user_suffix: str = None) -> dict:
+        """Generates the configuration in the format suitable for old versions of modern Antminers."""
```
Comment (b-rowan) [inline pyasic/config/__init__.py:64]: You can just put `...for old versions of Antminers.`

File: pyasic/config/__init__.py [hunk 10]
```diff
@@ -149,18 +165,22 @@ def from_am_modern(cls, web_conf: dict) -> "MinerConfig":
...
         return cls(pools=PoolConfig.from_am_modern(web_conf))
 
     @classmethod
     def from_inno(cls, web_pools: list) -> "MinerConfig":
+        """Constructs a MinerConfig object from web configuration for Innosilicon miners."""
         return cls(pools=PoolConfig.from_inno(web_pools))
 
     @classmethod
     def from_bosminer(cls, toml_conf: dict) -> "MinerConfig":
+        """Constructs a MinerConfig object from TOML configuration for BOSminer."""
```
Comment (b-rowan) [inline pyasic/config/__init__.py:183]: This constructs from the bosminer.toml file, same as the `to_bosminer` dumps a dict for writing to that file as toml.

**Review by b-rowan — CHANGES_REQUESTED** [review]
Some small changes, then it should be good to go.

File: pyasic/config/__init__.py (hunk 10 above)
Comment (1e9abhi1e10) [reply pyasic/config/__init__.py:183]: did you mean `as_bosminer` instead of `to_bosminer`?

Comment (b-rowan) [reply pyasic/config/__init__.py:183]: Yes, as_bosminer.

**Comment by b-rowan** [comment]
Good to merge?

**Review by UpstreamData — APPROVED** [review]
Should be good to go now.

**Pull request merged**